src/shared/cloudformation/templateRegistry.ts

```diff
@@ -21,6 +23,22 @@ export class CloudFormationTemplateRegistry {
         this.templateRegistryData = new Map<string, CloudFormation.Template>()
     }
 
+    /**
+     * Normalizes `path` for storage/lookup:
+     * - coerces to an absolute path, if the path is relative and the current
+     *   active document has a workspace
+     * - pipes through `pathutils.normalize()`
+     */
+    private normalize(path: string): string {
+        if (!path_.isAbsolute(path)) {
```

Comment:
At what points will a file touch the registry while the path isn't relative? I'm wary about using `vscode.window.activeTextEditor?.document.uri` since a lot of the registry's job is scraping, which means there's no guarantee that the active window is in the same workspace folder as the file that's being added.

Reply:
yeah this is probably not the right approach, instead we should throw if relative path is given to the registry. Or make this behavior optional, so that the caller can say "I want relative path to be forced to an absolute path".

Problem is that if e.g. launch.json has a relative path, full path is needed to resolve the template. That coercion should be done at the caller, yes, but unfortunately we end up needing to query `activeTextEditor` unless we pass the workspace folder around in many more places.

Reply:
updated